x-pack/plugins/observability_solution/infra/public/components/shared/templates/infra_page_template.tsx

```diff
@@ -50,10 +54,19 @@ export const InfraPageTemplate = ({
     });
   });
 
+  const { isEntityCentricExperienceEnabled } = useEntityCentricExperienceSetting();
+  const { asset } = useAssetDetailsRenderPropsContext();
+  const { dataStreams, status: entitySummaryStatus } = useEntitySummary({
+    entityType: asset?.type,
+    entityId: asset?.id,
+  });
```

Comment:
I noticed that when the Asset Details already calls `/summary` 2 times, this adds a 3rd call to the flow

<img width="487" alt="image" src="https://github.com/user-attachments/assets/14b69335-8ab6-49c4-a921-ca59125d7d11">

We need to optmize this.

>[!NOTE]
>I've also noticed that the `/summary` endpoint is called whenever the `overview`, `metrics`, and `processes` tabs are opened.  Ideally, we should make a single call to this endpoint and store the results in a context, but this can be addressed in a separate PR.

I would suggest the following (This is just an idea, what's most important is avoiding the 3rd call to the mounting flow):

On `infra/public/components/asset_details/template/page.tsx` (This will make the change affect only the asset details, which is what we want)
 - call `useEntityCentricExperienceSetting()`;
 - On `InfraPageTemplate`, pass `undefined` to `onboardingFlow` if EEM conditions are met. eg:
 ```ts
  const hasEEM =
    (isEntityCentricExperienceEnabled && isLogsSignal(dataStreams)) || isMetricsSignal(dataStreams);

  return (
    <InfraPageTemplate
      onboardingFlow={
        isPending(entitySummaryStatus) || hasEEM
          ? undefined
          : asset.type === 'host'
          ? OnboardingFlow.Hosts
          : OnboardingFlow.Infra
      }
```

On `infra/public/components/shared/templates/infra_page_template.tsx`
- Leave `onboardingFlow` as optional, removing the default `OnboardingFlow.Infra` value
- Only call `/api/metrics/source/hasData` if `onboardingFlow` is not `undefined`

On `infra/public/components/shared/templates/no_data_config.ts`
- Make `onboardingFlow` optional and early return if it's `undefined`

Reply:
Hey @crespocarlos thanks a lot for the suggested changes.

I applied your proposal in https://github.com/elastic/kibana/pull/196387/commits/8bf92b45a9f3d8b12a6762665a8f456dc8039830 and https://github.com/elastic/kibana/pull/196387/commits/92595fbbdd747fb38504b3a6f8e840888b4506fb.

Regarding this:

> I've also noticed that the /summary endpoint is called whenever the overview, metrics, and processes tabs are opened. Ideally, we should make a single call to this endpoint and store the results in a context, but this can be addressed in a separate PR.

I completely agree with making better performance decisions. I’d love to work on this improvement in a separate issue/PR.

Reply:
Thanks for creating the ticket :).